src/Symfony/Component/DependencyInjection/Compiler/ResolveTaggedIteratorArgumentPass.php

```diff
@@ -28,7 +28,12 @@ protected function processValue(mixed $value, bool $isRoot = false): mixed
             return parent::processValue($value, $isRoot);
         }
 
-        $value->setValues($this->findAndSortTaggedServices($value, $this->container));
+        $exclude = $value->getExclude();
+        if ($value->autoExcludeReferencingService() && !\in_array($this->currentId, $exclude, true) && $this->container->getDefinition($this->currentId)->hasTag($value->getTag())) {
```

Comment:
we don't really need the && parts, isn't it?

Reply:
Yes, we don't. It just felt better to avoid putting duplicated entries and/or irrelevant services as part in the `exclude` list passed to `findAndSortTaggedServices()`. Now removed